### PR TITLE
address dev package install ordering

### LINKED ISSUE #202: `GH-R-CMD-CHECK` workflow overwrites branch dev installs
GH-R-CMD-CHECK uses `try(pak::pkg_install())` for each dev package:

https://github.com/tidymodels/extratests/blob/def7031b67ba1c615e3c3fe12f15d6edb1cc3cd7/.github/workflows/GH-R-CMD-check.yaml#L61-L83

This is nice in that we don't have to deal with known dependency resolution issues, but also means that installs can be overwritten by later ones, e.g. in https://github.com/tidymodels/extratests/pull/198, making it difficult to test against dev branches. 

We could:

1) Try to install in an order that minimizes overwritten installs, and/or
2) Allow package dependency resolution for core packages by passing them as a vector to `pak::pkg_install()`, like:

```r
try(pak::pkg_install(paste0("tidymodels/", c("rsample", "recipes", "parsnip", "dials", "workflows", "tune"))))
```

This would mean that we're alerted when packages have circular dev dependencies, which is nice in the case where we did that accidentally but would require making temporary changes every time we do this knowingly.

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #202. 

In the cases where we have known circular dependencies, we can just pull the offending packages back into their own `pkg_install()` lines temporarily.

### CODE REVIEW

**Comment by simonpcouch** [comment]
CRAN failures are related to #209.

File: .github/workflows/GH-R-CMD-check.yaml [hunk 1]
```diff
@@ -58,30 +58,38 @@ jobs:
...
+            "tidymodels/baguette",
+            "tidymodels/bonsai",
+            "tidymodels/censored",
+            "tidymodels/discrim",
+            "tidymodels/multilevelmod",
+            "tidymodels/parsnip",
+            "tidymodels/plsmod",
+            "tidymodels/poissonreg",
+            "tidymodels/rules",
+            "tidymodels/themis"
```
Comment (hfrick) [inline .github/workflows/GH-R-CMD-check.yaml:83]: themis is a recipes extension, should it go in one block with recipes? or update the comment?

**Review by hfrick — APPROVED** [review]
Love it! 🙌

File: .github/workflows/GH-R-CMD-check.yaml (hunk 1 above)
Comment (simonpcouch) [reply .github/workflows/GH-R-CMD-check.yaml]: woops, thank you!

**Pull request merged**